sample-macos-app/sample-macos-app/ImageLoading.swift

```diff
@@ -0,0 +1,36 @@
+import Cocoa
+
+public extension NSImageView {
+    private var activeTask: URLSessionDataTask? {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKey.task) as? URLSessionDataTask
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKey.task, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+   
+    func loadImageByUrl(url: String, completion: @escaping () -> Void) {
+        cancelImageLoading()
+
+        let task = URLSession.shared.dataTask(with: URL(string: url)!) { [weak self] data, d, error in
+            guard let data = data, error == nil else { return }
```

Comment:
Is it okay that the completion will not be called if guard statements are not satisfied? Usually, it is not. We have to notify the caller that the task has finished.

Reply:
Fixed